Framework/DataObjects/inc/MantidDataObjects/BasePeak.h

```diff
@@ -0,0 +1,197 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidGeometry/Crystal/IPeak.h"
+#include "MantidGeometry/Crystal/PeakShape.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/Matrix.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/System.h"
+#include "MantidKernel/V3D.h"
+#include <boost/optional.hpp>
+#include <memory>
+
+namespace Mantid {
+
+namespace Geometry {
+class InstrumentRayTracer;
+}
+
+namespace DataObjects {
+
+/** Structure describing a single-crystal peak. This is a partial
+ * implementation of IPeak and should contain everything that is common
+ * to Peak and LeanPeak.
+ *
+ */
+class DLLExport BasePeak : public Geometry::IPeak {
+public:
+  /// Allow PeakColumn class to directly access members.
+  friend class PeakColumn;
+
+  BasePeak();
+  BasePeak(const Mantid::Kernel::Matrix<double> &goniometer);
+  /// Copy constructor
+  BasePeak(const BasePeak &other);
+
+  // MSVC 2015/17 can build with noexcept = default however
+  // intellisense still incorrectly reports this as an error despite compiling.
+  // https://connect.microsoft.com/VisualStudio/feedback/details/1795240/visual-c-2015-default-move-constructor-and-noexcept-keyw
+  // ord-bug For that reason we still use the supplied default which should be
+  // noexcept once the above is fixed we can remove this workaround
+#if defined(_MSC_VER) && _MSC_VER <= 1910
+  BasePeak(BasePeak &&) = default;
+  BasePeak &operator=(BasePeak &&) = default;
+#elif ((__GNUC__ < 4) || (__GNUC__ == 4 && __GNUC_MINOR__ <= 8))
```

Comment:
This `#elif` is unnecessary. We're using gcc7 as the oldest allowed.

Reply:
:+1: Yeah this was really just copied from `Peak.h`, so I can remove it some all three places.